### PR TITLE
SPECIES NUKING 2023: TOTAL EXTERNAL ORGAN DEATH

### PR DESCRIPTION
**Pull request by ChungusGamer666**
## About The Pull Request

Ports https://github.com/DaedalusDock/daedalusdock/pull/322, kind of
image

Fixes https://github.com/tgstation/tgstation/issues/76353
Fixes https://github.com/tgstation/tgstation/issues/76599
Fixes https://github.com/tgstation/tgstation/issues/76700

**Mushroom caps, fox ears and cat ears now use this new system for their overlays.** 

## Why It's Good For The Game

Just take a look at eyes and ears to truly appreciate how bad the situation gets as soon as internal organs have to draw an overlay on the owner. Moving external organ behavior down to bodyparts will greatly facilitate this kind of behavior.

## Changelog

:cl:
refactor: Organ code has been refactored in major ways. Report any organ related bugs on the github.
del: Removed feature manipulation surgeries due to aforementioned organ refactor.
/:cl:

### CODE REVIEW

**Comment by MrMelbert** [comment]
Don't merge this without a few test merges at the very least, organs-in-bodyparts historically has had some very nasty side effects

**Comment by ChungusGamer666** [comment]
This PR will 100% cause a metric fuckton of issues

**Comment by Time-Green** [comment]
Hopefully shouldn't be too crazy? The drawing behaviour has already been moved to bodypart_overlays, it's just the interface that needs moving down

I'm more worried about putting organs in limbs, since this is historically what caused the first biblical plagues

**Comment by Wollywoger** [comment]
i JUST got (mostly, like 90%) done with a project on a downstream that uses external organs heavily

i hope to god you are moving most of the workable code from external organs to their "internal" counter parts, because i tried to make a custom draws-on-mob-visually ears and immediately wanted to refactor the entire system (note: I am terrible at coding to begin with)

I will echo time-green's statements, doubly so because there is no "both arms" or "both legs" storage in the codebase

**Comment by ChungusGamer666** [comment]
@Wollywoger Pretty much all external organ behavior will be moved down to organ, so, don't worry too much

**Comment by ChungusGamer666** [comment]
as expected brains are the big fucking issue with moving organs to bodyparts, truly epic

**Comment by ChungusGamer666** [comment]
> as expected brains are the big fucking issue with moving organs to bodyparts, truly epic

it works (?)
image

**Comment by ChungusGamer666** [comment]
PREPARE FOR TROUBLE AND MAKE IT DOUBLE
HERE COMES THE MOTHER OF ALL FILES CHANGED

**Comment by ChungusGamer666** [comment]
image

**Comment by ChungusGamer666** [comment]
image

**Comment by ChungusGamer666** [comment]
of course we got a merge conflict in this bitch already

**Comment by ChungusGamer666** [comment]
image
if icondiffbot screams at me i will have a stroke

**Comment by Jolly-66** [comment]
Cries in having to redo species downstream for the nth time 

I solemnly hope it won't be a massive pain in the ass to converting species with external organs over to this new system.

Why did I EVER add so many species with external organs.

**Comment by ChungusGamer666** [comment]
screenshot tests apparently don't support RESET_COLOR
i am going to fucking scream

**Comment by ChungusGamer666** [comment]
finally
working felinid external ears

**Comment by ChungusGamer666** [comment]
IT IS... DONE...

**Comment by ChungusGamer666** [comment]
> IT IS... DONE...

I'M MAKING IT HARDER ON MYSELF BY TRYING TO FIX MONKEY TAILS

**Comment by ChungusGamer666** [comment]
the brain damage is starting to seep in

**Comment by ChungusGamer666** [comment]
image
the hard work finally pays off

**Comment by ChungusGamer666** [comment]
I spent actual hours debugging actual fucking monkey tails
My soul has been entirely drained out of my body

**Comment by ChungusGamer666** [comment]
IT IS DONE (AGAIN)

**Comment by nikothedude** [comment]
> merge conflict

THE RIDE NEVER ENDS BABY

**Comment by Time-Green** [comment]
guess what?

**Comment by Time-Green** [comment]
You can wait with fixing the conflicts untill after I do my review to save you the workload. It'll probably be a few days before I get to it

**Comment by ChungusGamer666** [comment]
> 

got it, just gonna fix one last thing (in this case being HARS)

**Comment by Sealed101** [comment]
babe wake up new tail issue just dropped
- #76599

**Comment by ChungusGamer666** [comment]
> babe wake up new tail issue just dropped
> 
> * [Spawned Tails Randomly Decide Whether to Have a Bodypart sprite_datum or Not #76599](https://github.com/tgstation/tgstation/issues/76599)

Another one to fix in this PR I guess

**Comment by san7890** [comment]
draft until #76615 is addressed

**Comment by ChungusGamer666** [comment]
https://github.com/tgstation/tgstation/issues/76615 is fixed

File: code/modules/surgery/organs/internal/eyes/_eyes.dm
Comment (Time-Green) [inline code/modules/surgery/organs/internal/eyes/_eyes.dm:23]: You even did eyes??? Great stuff

File: code/modules/surgery/organs/internal/liver/_liver.dm
Comment (Time-Green) [inline code/modules/surgery/organs/internal/liver/_liver.dm:164]: mood

**Review by Time-Green — CHANGES_REQUESTED** [review]
This took a very long time to review, but tbf would've taken even longer if I had to code it myself. 

Amazing work! This is just a code review, I'm going to test this pretty hard before it gets merged to make sure the playerbase doesn't melt (I'll let you know when I get to it)

File: code/datums/sprite_accessories.dm
Comment (ChungusGamer666) [reply code/datums/sprite_accessories.dm:1772]: I had to do this due to sprite accessory fuckery, trust me.

File: code/modules/mob/living/simple_animal/friendly/cat.dm
Comment (ChungusGamer666) [reply code/modules/mob/living/simple_animal/friendly/cat.dm:23]: It appears this was an accidental removal.

File: code/modules/mob/living/simple_animal/friendly/cat.dm
Comment (ChungusGamer666) [reply code/modules/mob/living/simple_animal/friendly/cat.dm:74]: It appears this was an accidental removal.

File: code/modules/surgery/bodyparts/bodypart_visual.dm
Comment (ChungusGamer666) [reply code/modules/surgery/bodyparts/bodypart_visual.dm:4]: Sure.

File: code/modules/surgery/bodyparts/robot_bodyparts.dm
Comment (ChungusGamer666) [reply code/modules/surgery/bodyparts/robot_bodyparts.dm:17]: Defined on parent. Literally the same.

File: code/modules/surgery/organs/external/tails.dm
Comment (ChungusGamer666) [reply code/modules/surgery/organs/external/tails.dm:109]: As I said before, due to stupid sprite accessory code, this was necessary.

File: code/modules/surgery/organs/internal/eyes/_eyes.dm
Comment (ChungusGamer666) [reply code/modules/surgery/organs/internal/eyes/_eyes.dm:23]: No, eyes haven't been reworked, but will be reworked in the future.

File: code/modules/surgery/organs/internal/liver/_liver.dm
Comment (ChungusGamer666) [reply code/modules/surgery/organs/internal/liver/_liver.dm:164]: why was it plural :(

Comment (ChungusGamer666) [reply code/modules/surgery/organs/internal/liver/_liver.dm:164]: ah FUCK

File: code/modules/explorer_drone/exploration_events/trader.dm
Comment (ChungusGamer666) [reply code/modules/explorer_drone/exploration_events/trader.dm:92]: uh sure

File: code/modules/surgery/organ_manipulation.dm
Comment (Time-Green) [inline code/modules/surgery/organ_manipulation.dm:27]: I do think this is a bad change, the point is to let people easily change cosmetic organs.  It's fine for now, at least it's one less thing that can break and it can be readded later if demand is there

Comment (ChungusGamer666) [reply code/modules/surgery/organ_manipulation.dm:27]: There is not a foolproof way for me to identify "external" organs and therefore it's hard for me to keep this in.

Comment (ChungusGamer666) [reply code/modules/surgery/organ_manipulation.dm:27]: Also this is literally the same step, you can't even like, restyle the organs (?)

**Comment by Time-Green** [comment]
Mush cap organ doesnt work lol
image

Couldn't find any bugs messing around otherwise. I'll see if we can get it testmerged for a bit longer and then I'm convinced

**Comment by ChungusGamer666** [comment]
> Mush cap organ doesnt work lol image
> 
> Couldn't find any bugs messing around otherwise. I'll see if we can get it testmerged for a bit longer and then I'm convinced

To be fixed!!!

File: code/modules/surgery/organ_manipulation.dm
Comment (Time-Green) [reply code/modules/surgery/organ_manipulation.dm:27]: Nah it's way shorter than normal organ manip

**Comment by ChungusGamer666** [comment]
image
@Time-Green guess what

**Comment by Time-Green** [comment]
> image @Time-Green guess what

no way

**Comment by ChungusGamer666** [comment]
worked more kinks out, should be ready for a testmergin'

**Comment by ChungusGamer666** [comment]
@Time-Green @TheVekter please get another testmerge going soon, i want to finish this up so i can keep cookin' up more mob code improvements

**Comment by TheVekter** [comment]
> @Time-Green @TheVekter please get another testmerge going soon, i want to finish this up so i can keep cookin' up more mob code improvements

I requested it; I'll let you know how it goes.

**Comment by ChungusGamer666** [comment]
merge conflicts my behated

**Comment by ChungusGamer666** [comment]
ill fix this soonish

**Comment by ChungusGamer666** [comment]
there is no merge conflict anymore shithub, calm down

**Comment by TheVekter** [comment]
oh damn, the wordless oranges close

**Comment by Time-Green** [comment]
Well fuck

**Comment by kinneb** [comment]
the slice was scarfered

**Comment by TheVekter** [comment]
The big chungus is no longer among us

Hopefully someone who actually tests their code will come back and try this some day

**Comment by Time-Green** [comment]
> The big chungus is no longer among us
> 
> Hopefully someone who actually tests their code will come back and try this some day

I'll do it at some point (and in normal PR sizes since no one except me is going to review a 3500 line organ PR)

**Pull request closed without merging**
